src/PerspectiveLanding.elm

```diff
@@ -0,0 +1,178 @@
+module PerspectiveLanding exposing (..)
+
+import Api
+import Definition.Doc as Doc
+import Definition.Readme as Readme
+import Definition.Reference exposing (Reference)
+import Env exposing (Env)
+import FullyQualifiedName as FQN exposing (FQN)
+import Html exposing (Html, a, article, div, h2, header, p, section, span, strong, text)
+import Html.Attributes exposing (class, href, id, rel, target)
+import Namespace exposing (Namespace(..))
+import Perspective
+import RemoteData exposing (RemoteData(..))
+import UI
+import UI.Button as Button exposing (Button)
+import UI.Icon as Icon
+
+
+type alias Model =
+    Doc.DocFoldToggles
+
+
+init : Model
+init =
+    Doc.emptyDocFoldToggles
+
+
+type Msg
+    = OpenReference Reference
+    | ToggleDocFold Doc.FoldId
+    | Find
+
+
+type OutMsg
+    = OpenDefinition Reference
+    | ShowFinderRequest
+    | None
+
+
+update : Msg -> Model -> ( Model, OutMsg )
+update msg model =
+    case msg of
+        OpenReference r ->
+            ( model, OpenDefinition r )
+
+        Find ->
+            ( model, ShowFinderRequest )
+
+        ToggleDocFold fid ->
+            ( Doc.toggleFold model fid, None )
+
+
+
+-- VIEW
+
+
+container : List (Html msg) -> Html msg
+container content =
+    article [ id "perspective-landing" ]
+        [ section
+            [ id "perspective-landing-content" ]
+            content
+        ]
+
+
+viewLoading : Html msg
+viewLoading =
+    container
+        [ div
+            [ class "loading" ]
+            [ UI.loadingPlaceholderRow
+            , UI.loadingPlaceholderRow
+            ]
+        ]
+
+
+viewError : FQN -> String -> Html msg
+viewError fqn message =
+    container
+        [ div
+            [ class "perspective-landing-error" ]
+            [ header [] [ Icon.view Icon.warn, text ("Error loading " ++ FQN.toString fqn) ]
+            , p [] [ text message ]
+            ]
+        ]
+
+
+viewEmptyState : Html msg -> List (Html msg) -> Button msg -> Html msg
+viewEmptyState title description cta =
```

Comment:
All empty states happen when no definitions are open and thus within PerspectiveLanding. In the future it'd be nice to have EmptyState be a more common design system component.